cluster_orchestrator/cluster-scheduler/calculation.py

```diff
@@ -11,7 +11,12 @@ def calculate(app, job: dict) -> Union[dict, NegativeSchedulingStatus]:
 
     # check here if job has any user preferences, e.g. on a specific node, cpu architecture,
     constraints = job.get("constraints")
-    if constraints is not None and len(constraints) > 0:
+    if (
+        constraints is not None
+        and len(constraints) > 0
+        # The clusters constraint only plays a role in the cloud-scheduler.
+        and not (len(constraints) == 1 and constraints[0].get("type") == "clusters")
```

Comment:
since constraints list is growing, can we find a slightly better way? E.g. supported_constraints constant list and then something like:

```
if set(constraints) & set(supported_constraints):
 constraint_based_scheduing(...)
```

Reply:
Good point. We need to extract and perform this check on the constraint types and not on the entire constraint dics, because one cannot convert them to a set easily AFAIK. 
![image](https://github.com/oakestra/oakestra/assets/65814168/4f64440d-448d-44f8-89df-6a3e0764686b)